src/gfx/reverse.cpp

```diff
@@ -171,13 +171,24 @@ void reverse() {
 			        options.nbPalettes);
 		}
 
-		if (options.palSpecType == Options::EXPLICIT && palettes != options.palSpec) {
-			warning("Colors in the palette file do not match those specified with `-c`!");
+		if (options.palSpecType == Options::EXPLICIT) {
+			for (auto& palette : palettes) {
+				for (auto& specPalette : options.palSpec) {
+					for (auto& color : palette) {
+						for (auto& specColor : specPalette) {
+							if (!specColor || color != specColor.value()) {
```

Comment:
Hmm, warning if `!specColor` means that any `-r N -c '...,#none,...'` will warn... shouldn't this be

```suggestion
							if (specColor && color != specColor.value()) {
```

?